src/lib/core/components/visualizations/implementations/BoxplotVisualization.tsx

```diff
@@ -0,0 +1,403 @@
+          <InputVariables
+            inputs={[
+              {
+                name: 'xAxisVariable',
+                label: 'X-axis',
+              },
+              {
+                name: 'yAxisVariable',
+                label: 'Y-axis',
+              },
+              {
+                name: 'overlayVariable',
+                label: 'Overlay (Optional)',
+              },
+              {
+                name: 'facetVariable',
+                label: 'Facet (Optional)',
+              },
+            ]}
+            entities={entities}
+            values={{
+              xAxisVariable: vizConfig.xAxisVariable,
+              yAxisVariable: vizConfig.yAxisVariable,
+              overlayVariable: vizConfig.overlayVariable,
+            }}
+            onChange={handleInputVariableChange}
+            constraints={dataElementConstraints}
+            dataElementDependencyOrder={dataElementDependencyOrder}
+            starredVariables={starredVariables}
+            toggleStarredVariable={toggleStarredVariable}
+          />
+        </div>
+      )}
+
+      {data.error && fullscreen && (
+        <div
+          style={{
+            fontSize: '1.2em',
+            padding: '1em',
+            background: 'rgb(255, 233, 233) none repeat scroll 0% 0%',
+            borderRadius: '.5em',
+            margin: '.5em 0',
+            color: '#333',
+            border: '1px solid #d9cdcd',
+            display: 'flex',
+          }}
+        >
+          <i className="fa fa-warning" style={{ marginRight: '1ex' }}></i>{' '}
+          {data.error instanceof Error
+            ? data.error.message
+            : String(data.error)}
+        </div>
+      )}
+      {fullscreen ? (
+        <BoxplotWithControls
+          // data.value
+          data={data.value && !data.pending ? data.value.series : []}
+          containerStyles={{
+            width: '100%',
+            height: 450,
+          }}
+          // title={'boxplot'}
+          orientation={'vertical'}
+          // orientation={'horizontal'}
+          // add condition to show legend when overlayVariable is used
+          displayLegend={
+            data.value &&
+            (data.value.series.length > 1 || vizConfig.overlayVariable != null)
+          }
+          independentAxisLabel={
+            findVariable(vizConfig.xAxisVariable)?.displayName
+          }
+          dependentAxisLabel={
+            findVariable(vizConfig.yAxisVariable)?.displayName
+          }
+          // show/hide independent/dependent axis tick label
+          showIndependentAxisTickLabel={true}
+          showDependentAxisTickLabel={true}
+          showMean={true}
+          interactive={true}
+          showSpinner={data.pending}
+          // this is required for date type
+          // not sure why enrollment year's type is number, not date
+          dependentValueType={findVariable(vizConfig.yAxisVariable)?.type}
```

Comment:
I think we're going to remove all date handling from boxplot, so `dependentValueType` can probably also be removed.

Note that "enrollment year" is not really our problem.  We can add y-axis range controls so users can fix it, maybe.

Reply:
yeah since no 'date' data at dependentValueType is expected, the prop can be removed at viz

Reply:
deleted `dependentValueType` prop at viz